src/components/calculator.js

```diff
@@ -8,59 +8,54 @@ import calculate from '../logic/calculate';
  ,jsx-a11y/no-static-element-interactions
  ,react/prop-types
  ,react/no-unused-state */
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: null,
-      next: null,
-      operation: null,
-    };
-    this.performClick = this.performClick.bind(this);
-  }
+let [obj, setObj] = [{}, () => {}];
 
-  performClick(e) {
-    const value = e.target.innerHTML;
-    const { total, next, operation } = calculate(this.state, value);
-    this.setState({ total, next, operation });
-    const screen = document.querySelector('.screen');
-    if (next !== null) {
-      screen.innerHTML = next;
-    } else if (total !== null) {
-      screen.innerHTML = total;
-    } else if (next !== null && total !== null) {
-      screen.innerHTML = total;
-    } else {
-      screen.innerHTML = '0';
-    }
+const performClick = (e) => {
+  const value = e.target.innerHTML;
+  const { total, next, operation } = calculate(obj, value);
+  setObj({ total, next, operation });
+  const screen = document.querySelector('.screen');
+  if (next !== null) {
+    screen.innerHTML = next;
+  } else if (total !== null) {
+    screen.innerHTML = total;
+  } else if (next !== null && total !== null) {
+    screen.innerHTML = total;
+  } else {
+    screen.innerHTML = '0';
   }
+};
```

Comment:
- [ ] Kindly place this event handler inside your Calculator component/function. For hook-based functions, event handlers for the component are placed inside the main function which is `Calculator` in your case